ZCRMiOS/Classes/DashBoardAPIHandler.swift

```diff
@@ -0,0 +1,1737 @@
+                
+            case metaComponentAPINames.componentXPosition:
+                xPosition = Int((value as? String) ?? "default")
+                
+            case metaComponentAPINames.componentYPosition:
+                yPosition = Int((value as? String) ?? "default")
+                
+            default:
+                print("UNKNOWN KEY \(key) ENCOUNTERED IN DASHBOARD META COMPONENT LAYOUT PARSING !!")
+                
+            }
+            
+        }
+        
+        return (width: width, height: height, xPosition: xPosition, yPosition: yPosition)
+    }
+    
+    
+} // Extension ends
+
+
+
+extension DashBoardAPIHandler {
+    
+    //MARK:- *** DASHBOARD COMPONENT PARSERS ***
+    
+    
+    internal func getDashBoardComponentFrom(_ componentJSON: [String:Any]) -> ZCRMDashBoardComponent {
+        
+        let componentObj = ZCRMDashBoardComponent()
+        
+        for(key,value) in componentJSON {
+            
+            switch(key) {
+                
+            case componentAPINames.componentName:
+                componentObj.setComponent(Name: value as? String)
+                
+            case componentAPINames.componentCategory:
+                let category = CompCategory(rawValue: (value as? String) ?? "default")
+                componentObj.setComponent(Category: category)
+                
+            case componentAPINames.componentMarker:
+                let arrayOfComponentMarkersObj = getComponentMarkersFrom(componentJSON)
+                componentObj.setComponent(Markers: arrayOfComponentMarkersObj)
+                
+            case componentAPINames.componentProps:
+                setComponentPropertiesFor(componentObj, Using: componentJSON)
+                
+            case componentAPINames.reportID:
+                componentObj.setReport(ID: value as? Int64)
+                
+            case componentAPINames.componentChunks:
+                setComponentChunksValues(To: componentObj, Using: value as? [[String:Any]])
+                
+            default:
+                print("UNKNOWN KEY \(key) ENCOUNTERED IN COMPONENT PARSING")
+                
+            } // switch ends
+            
+        } // loop ends
+        
+        return componentObj
+    } // func ends
+    
+    
+    
+    
+    internal func getComponentMarkersFrom(_ componentJSON:[String:Any]) -> [ComponentMarkers] {
+        
+        
+        let ArrayOfCompMarkerJSON = componentJSON.getArrayOfDictionaries(key: componentAPINames.componentMarker)
+        
+        var ArrayOfComponentMarkerObj = [ComponentMarkers]()
+        
+        var x:String? // can be User ID (Int64) or pickListValue (String)
+        var y:Int?
+        
+        
+        for compMarkersJSON  in ArrayOfCompMarkerJSON {
+            
+            for (key,value) in compMarkersJSON {
+                
+                switch (key) {
+                    
```

Comment:
Why didn't you use our dictionary extension?
Remove empty lines

Reply:
https://github.com/zoho/CRM-iOSSDK/pull/15#discussion_r214515160